Can't build without the clipboard feature

First of all, thanks for bringing egui to bevy !

It seems that the `manage_clipboard` isn't considered in several places. I tried:
```
cargo run --example simple --no-default-features --features="bevy/x11 bevy/bevy_wgpu"
```
And got a bunch of errors:

<details>

```
error[E0432]: unresolved import `crate::EguiClipboard`
 --> src/systems.rs:2:5
  |
2 |     EguiClipboard, EguiContext, EguiInput, EguiOutput, EguiSettings, EguiShapes, WindowSize,
  |     ^^^^^^^^^^^^^ no `EguiClipboard` in the root

error[E0433]: failed to resolve: use of undeclared type `ClipboardContext`
   --> src/lib.rs:185:17
    |
185 |                 ClipboardContext::new()
    |                 ^^^^^^^^^^^^^^^^ use of undeclared type `ClipboardContext`

error[E0433]: failed to resolve: use of undeclared type `RefCell`
   --> src/lib.rs:186:26
    |
186 |                     .map(RefCell::new)
    |                          ^^^^^^^ use of undeclared type `RefCell`

error[E0433]: failed to resolve: use of undeclared type `EguiClipboard`
   --> src/lib.rs:349:38
    |
349 |         resources.get_or_insert_with(EguiClipboard::default);
    |                                      ^^^^^^^^^^^^^ use of undeclared type `EguiClipboard`

error[E0412]: cannot find type `EguiClipboard` in this scope
   --> src/lib.rs:140:6
    |
140 | impl EguiClipboard {
    |      ^^^^^^^^^^^^^ not found in this scope

error[E0412]: cannot find type `RefMut` in this scope
   --> src/lib.rs:182:29
    |
182 |     fn get(&self) -> Option<RefMut<ClipboardContext>> {
    |                             ^^^^^^ not found in this scope
    |
help: consider importing one of these items
    |
49  | use bevy::ecs::RefMut;
    |
49  | use core::cell::RefMut;
    |
49  | use std::cell::RefMut;
    |

error[E0412]: cannot find type `ClipboardContext` in this scope
   --> src/lib.rs:182:36
    |
140 | impl EguiClipboard {
    |     - help: you might be missing a type parameter: `<ClipboardContext>`
...
182 |     fn get(&self) -> Option<RefMut<ClipboardContext>> {
    |                                    ^^^^^^^^^^^^^^^^ not found in this scope

warning: unused import: `log`
  --> src/systems.rs:13:5
   |
13 |     log,
   |     ^^^
   |
   = note: `#[warn(unused_imports)]` on by default

error[E0599]: no method named `system` found for fn item `for<'r, 's, 't0, 't1, 't2, 't3> fn(Res<'r, EguiContext>, ResMut<'s, EguiOutput>, ResMut<'t0, EguiShapes>, ResMut<'t1, [type error]>, Res<'t2, bevy::prelude::Windows>, Res<'t3, WinitWindows>) {systems::process_output}` in the current scope
   --> src/lib.rs:342:69
    |
342 |         app.add_system_to_stage(stage::UI_FRAME_END, process_output.system());
    |                                                                     ^^^^^^ method not found in `for<'r, 's, 't0, 't1, 't2, 't3> fn(Res<'r, EguiContext>, ResMut<'s, EguiOutput>, ResMut<'t0, EguiShapes>, ResMut<'t1, [type error]>, Res<'t2, bevy::prelude::Windows>, Res<'t3, WinitWindows>) {systems::process_output}`
    |
    = note: `process_output` is a function, perhaps you wish to call it

error: aborting due to 8 previous errors; 1 warning emitted
```

</details>

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -31,10 +31,10 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
         features: [
-            '',
-            'manage_clipboard',
-            'open_url',
-            'manage_clipboard,open_url',
+            'bevy/x11,bevy/png,bevy/bevy_wgpu',
+            'bevy/x11,bevy/png,bevy/bevy_wgpu,manage_clipboard',
+            'bevy/x11,bevy/png,bevy/bevy_wgpu,open_url',
+            'bevy/x11,bevy/png,bevy/bevy_wgpu,manage_clipboard,open_url',
         ]
     steps:
       - uses: actions/checkout@v2
@@ -54,6 +54,11 @@ jobs:
           restore-keys: |
             cache-${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
             cache-${{ runner.os }}-cargo
+      - name: Work around MacOS + Cargo + Github Actions cache bug # Stolen from https://github.com/denoland/deno/commit/265c25754b56c60f72dd2a67cbb4c12696c7b3bb
+        if: runner.os == 'macOS'
+        uses: actions-rs/cargo@v1
+        with:
+          command: clean
       - name: Simple example
         uses: actions-rs/cargo@v1
         with:
@@ -96,13 +101,7 @@ jobs:
           restore-keys: |
             cache-wasm32-cargo-${{ hashFiles('**/Cargo.toml') }}
             cache-wasm32-cargo
-      - name: Simple example
-        uses: actions-rs/cargo@v1
+      - uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --example simple --no-default-features --target=wasm32-unknown-unknown --features=${{ matrix.features }} -- -D warnings
-      - name: UI example
-        uses: actions-rs/cargo@v1
-        with:
-          command: clippy
-          args: --example ui --no-default-features --target=wasm32-unknown-unknown --features=${{ matrix.features }} -- -D warnings
+          args: --no-default-features --target=wasm32-unknown-unknown --features=${{ matrix.features }} -- -D warnings

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -105,3 +105,28 @@ jobs:
         with:
           command: clippy
           args: --no-default-features --target=wasm32-unknown-unknown --features=${{ matrix.features }} -- -D warnings
+
+  doc:
+    name: Check documentation
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: cache-doc-cargo-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            cache-doc-cargo-${{ hashFiles('**/Cargo.toml') }}
+            cache-doc-cargo
+      - uses: actions-rs/cargo@v1
+        with:
+          command: doc
+          args: --all

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -29,7 +29,13 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        features: [
+            '',
+            'manage_clipboard',
+            'open_url',
+            'manage_clipboard,open_url',
+        ]
     steps:
       - uses: actions/checkout@v2
       - uses: actions-rs/toolchain@v1
@@ -48,7 +54,55 @@ jobs:
           restore-keys: |
             cache-${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
             cache-${{ runner.os }}-cargo
-      - uses: actions-rs/cargo@v1
+      - name: Simple example
+        uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --example ui --features=bevy/x11,bevy/png,bevy/bevy_wgpu -- -D warnings
+          args: --example simple --no-default-features --features=${{ matrix.features }} -- -D warnings
+      - name: UI example
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --example ui --no-default-features --features=${{ matrix.features }} -- -D warnings
+
+  clippy_wasm32:
+    name: Clippy check (wasm32)
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        features: [
+            '',
+            'manage_clipboard',
+            'open_url',
+            'manage_clipboard,open_url',
+        ]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+          components: clippy
+          target: wasm32-unknown-unknown
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: cache-wasm32-cargo-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            cache-wasm32-cargo-${{ hashFiles('**/Cargo.toml') }}
+            cache-wasm32-cargo
+      - name: Simple example
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --example simple --no-default-features --target=wasm32-unknown-unknown --features=${{ matrix.features }} -- -D warnings
+      - name: UI example
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --example ui --no-default-features --target=wasm32-unknown-unknown --features=${{ matrix.features }} -- -D warnings

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,0 +1,12 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [0.1.1] - 18-Jan-2020
+
+### Fixed
+
+- Fix compilation errors when no features are set.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,7 +20,7 @@ bevy = { version = "0.4", default-features = false, features = ["render"] }
 bevy_winit = "0.4.0"
 egui = "0.8.0"
 webbrowser = { version = "0.5.5", optional = true }
-winit = { version = "0.24.0", default-features = false }
+winit = { version = "0.24.0", features = ["x11"], default-features = false }
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
 clipboard = { version = "0.5.0", optional = true }
@@ -30,3 +30,6 @@ thread_local = { version = "1.1.0", optional = true }
 name = "ui"
 path = "examples/ui.rs"
 required-features = ["bevy/x11", "bevy/png", "bevy/bevy_wgpu"]
+
+[package.metadata.docs.rs]
+all-features = true

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "bevy_egui"
-version = "0.1.0"
+version = "0.1.1"
 authors = ["mvlabat <mvlabat@gmail.com>"]
 description = "A plugin for Egui integration into Bevy"
 license = "MIT"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -137,6 +137,7 @@ pub struct EguiClipboard {
     clipboard: String,
 }
 
+#[cfg(feature = "manage_clipboard")]
 impl EguiClipboard {
     /// Sets clipboard contents.
     pub fn set_contents(&mut self, contents: &str) {
@@ -331,12 +332,6 @@ impl Plugin for EguiPlugin {
             SystemStage::parallel(),
         );
 
-        #[cfg(all(
-            feature = "manage_clipboard",
-            target_arch = "wasm32",
-            web_sys_unstable_apis
-        ))]
-        app.add_startup_system(setup_clipboard_event_listeners.system());
         app.add_system_to_stage(stage::INPUT, process_input.system());
         app.add_system_to_stage(stage::UI_FRAME, begin_frame.system());
         app.add_system_to_stage(stage::UI_FRAME_END, process_output.system());
@@ -346,6 +341,7 @@ impl Plugin for EguiPlugin {
         resources.get_or_insert_with(EguiInput::default);
         resources.get_or_insert_with(EguiOutput::default);
         resources.get_or_insert_with(EguiShapes::default);
+        #[cfg(feature = "manage_clipboard")]
         resources.get_or_insert_with(EguiClipboard::default);
         resources.insert(EguiContext::new());
         resources.insert(WindowSize::new(0.0, 0.0, 0.0));

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -175,6 +175,7 @@ impl EguiClipboard {
     }
 
     #[cfg(target_arch = "wasm32")]
+    #[allow(clippy::unnecessary_wraps)]
     fn get_contents_impl(&self) -> Option<String> {
         Some(self.clipboard.clone())
     }

--- a/src/systems.rs
+++ b/src/systems.rs
@@ -1,6 +1,4 @@
-use crate::{
-    EguiClipboard, EguiContext, EguiInput, EguiOutput, EguiSettings, EguiShapes, WindowSize,
-};
+use crate::{EguiContext, EguiInput, EguiOutput, EguiSettings, EguiShapes, WindowSize};
 use bevy::{
     app::Events,
     core::Time,
@@ -19,7 +17,7 @@ use bevy_winit::WinitWindows;
 pub fn process_input(
     mut egui_context: ResMut<EguiContext>,
     mut egui_input: ResMut<EguiInput>,
-    #[cfg(feature = "manage_clipboard")] egui_clipboard: Res<EguiClipboard>,
+    #[cfg(feature = "manage_clipboard")] egui_clipboard: Res<crate::EguiClipboard>,
     ev_cursor: Res<Events<CursorMoved>>,
     ev_mouse_wheel: Res<Events<MouseWheel>>,
     ev_received_character: Res<Events<ReceivedCharacter>>,
@@ -154,7 +152,8 @@ pub fn process_output(
     egui_context: Res<EguiContext>,
     mut egui_output: ResMut<EguiOutput>,
     mut egui_shapes: ResMut<EguiShapes>,
-    mut egui_clipboard: ResMut<EguiClipboard>,
+    #[cfg(feature = "manage_clipboard")]
+    mut egui_clipboard: ResMut<crate::EguiClipboard>,
     windows: Res<Windows>,
     winit_windows: Res<WinitWindows>,
 ) {
@@ -170,7 +169,11 @@ pub fn process_output(
     if let Some(window) = windows.get_primary() {
         if let Some(winit_window) = winit_windows.get_window(window.id()) {
             winit_window.set_cursor_icon(egui_to_winit_cursor_icon(output.cursor_icon));
+        } else {
+            log::error!("No winit window found for the primary window");
         }
+    } else {
+        log::warn!("No primary window detected");
     }
 
     #[cfg(feature = "open_url")]

--- a/src/systems.rs
+++ b/src/systems.rs
@@ -152,8 +152,7 @@ pub fn process_output(
     egui_context: Res<EguiContext>,
     mut egui_output: ResMut<EguiOutput>,
     mut egui_shapes: ResMut<EguiShapes>,
-    #[cfg(feature = "manage_clipboard")]
-    mut egui_clipboard: ResMut<crate::EguiClipboard>,
+    #[cfg(feature = "manage_clipboard")] mut egui_clipboard: ResMut<crate::EguiClipboard>,
     windows: Res<Windows>,
     winit_windows: Res<WinitWindows>,
 ) {

